pandas/tests/indexing/test_indexing.py

```diff
@@ -627,6 +627,11 @@ def test_index_type_coercion(self):
                     idxr(s2)['0'] = 0
                     assert s2.index.is_object()
 
+    def test_empty_index_for_empty_dataframe(self):
+        empty_df = pd.DataFrame(columns=['A'])
```

Comment:
Can you do `index=pd.Index([], name='idx')`, to protect against a RangeIndex being automatically created in the future.